### PR TITLE
ELF: Fail layout when program header hooks return nullptr

### LINKED ISSUE #664: Trying to run /boot/kernel crashes the kernel
Running the command `/boot/kernel` at the command prompt results in this panic:
```
</boot/kernel> 

</boot/kernel> <(null)> 
Process: New process PID=17 with name=Shell
Thread{0x00336d10}: New thread TID=0 in Shell(17)
Closing fds in shell process:
Now we gotta wait on children:
  17641264
sys$waitpid(17, 0x010d2e38, 6)
Shell(17) do_exec(/boot/kernel): thread_count() = 1
VRA: Failed to allocate specific range: 10000(552960)
VRA: Failed to allocate specific range: 97000(28672)
0x00018d41  kmalloc_impl(unsigned long) +78
0x0002a2f2  AK::Vector<Range, 0>::ensure_capacity(int) [clone .part.18] +26
0x0002a5e4  RangeAllocator::deallocate(Range) +140
0x00028974  MemoryManager::unmap_region(Region&, bool) +162
0x0002906f  Region::~Region() +33
0x0001aa32  AK::NonnullOwnPtr<Region>::clear() [clone .isra.162] +20
0x0004fccf  AK::Vector<AK::NonnullOwnPtr<Region>, 0>::clear() +25
0x00021616  Process::do_exec(AK::String, AK::Vector<AK::String, 0>, AK::Vector<AK::String, 0>) +1796
0x00021a1c  Process::exec(AK::String, AK::Vector<AK::String, 0>, AK::Vector<AK::String, 0>) +324
0x00021c11  Process::sys$execve(char const*, char const**, char const**) +395
0x00025083  syscall_trap_entry +1216
0x00024b76  syscall_trap_handler +40
Shell(17) kmalloc(): PANIC! Out of memory (sucks, dude)
sum_free=0, real_size=104
```

Double-clicking the kernel from the file manager results in this panic:
```
Process: New process PID=18 with name=FileManager
Thread{0x0034d810}: New thread TID=0 in FileManager(18)
FileManager(18) do_exec(/boot/kernel): thread_count() = 1
VRA: Failed to allocate specific range: 10000(552960)
VRA: Failed to allocate specific range: 97000(28672)
FileManager(dump_backtrace() called from within itself, what the hell is going on!
FileManager(18) kmalloc(): PANIC! Out of memory (sucks, dude)
sum_free=0, real_size=24
```

### PR DESCRIPTION
**Pull request by ADKaster**
ElfLoader::layout has a "failed" variable that is never set.
This patch checks the return value of each hook (alloc/map section
and tls) and fails the load if they return null.

For some reason, I also needed to patch Kernel/Process.cpp so that
the alloc_section_hook and map_section_hook actually return nullptr
when allocating their regions fails? Not sure why vaddr.as_ptr()
wouldn't be nullptr when that call fails...

This should resolve #664 :)

### CODE REVIEW

**Comment by ADKaster** [comment]
Looks like clang-format decided to have its way with Kernel/Process.cpp. Not sure if that's kosher on a simple commit like this.

File: Kernel/Process.cpp [hunk 1]
```diff
@@ -2929,7 +2931,7 @@ int Process::sys$mount(const char* device_path, const char* mountpoint, const ch
 
     auto& mountpoint_custody = custody_or_error.value();
 
-    RefPtr<FS> fs { nullptr };
+    RefPtr<FS> fs{ nullptr };
```
Comment (awesomekling) [inline Kernel/Process.cpp]: This alignment is not consistent with the coding style. Are you using an old version of `clang-format` perhaps?

Comment (ADKaster) [reply Kernel/Process.cpp]: Yep that's what it looks like. Default clang-format on Ubuntu-18.04 was clang-format-6.0. clang-format-8.0 seems to like the style better.

**Comment by awesomekling** [comment]
Thanks for fixing this! :)

**Pull request merged**